api/catalog/api/views/media_views.py

```diff
@@ -211,6 +212,19 @@ def _thumbnail_proxy_comm(
             )
 
             return upstream_response, res_status, content_type
+        except requests.ReadTimeout as exc:
+            # Count the incident so that we can identify providers with most timeouts.
+            domain = urlparse(params["url"]).netloc
+            key = f"{settings.THUMBNAIL_TIMEOUT_PREFIX}{domain}"
+            try:
+                cache.incr(key)
+            except ValueError:  # Key does not exist.
+                cache.set(key, 1)
+
+            capture_exception(exc)
```

Comment:
Did we want to wrap this whole thing in a `try`/`except` to ensure that we handle any errors that might occur as part of `urlparse` or any other of these steps?

Reply:
very unlikely for URL parse to raise an exception. If we're making a request that times out, the URL would have to be valid and if so, `urlparse` should be able to get a `netloc` out of it. I tried it with some weird URLs I could come up with and `urlparse` can handle anything 😂.

Reply:
Fair point! Just being overly precautious 😄